### PR TITLE
Added ConsulRegistry

### PR DESCRIPTION
**Pull request by bprager**
- ConsulRegistry class added
- Test cases wired
- Parent POM updated to fixed Orbits Consul client

### CODE REVIEW

**Comment by lburgazzoli** [comment]
Some notes:
- can you please squash the commits in a single one ?
- please run mvn clean compile -Psourcecheck on camel-consul, there are quite a few checkstyle violations mostly because of tab vs space ;)

File: parent/pom.xml [hunk 1]
```diff
@@ -131,8 +131,8 @@
     <commons-pool2-version>2.4.2</commons-pool2-version>
     <commons-vfs2-version>2.0</commons-vfs2-version>
     <compress-lzf-version>1.0.3</compress-lzf-version>
-    <consul-client-version>0.12.0</consul-client-version>
-    <consul-client-bundle-version>0.12.0_1</consul-client-bundle-version>
+    <consul-client-version>0.12.2</consul-client-version>
+    <consul-client-bundle-version>0.12.2</consul-client-bundle-version>
```
Comment (lburgazzoli) [inline parent/pom.xml]: This should be 0.12.2_1 it is an OSGi bundle made by Service Mix (not yet released indeed)

**Comment by davsclaus** [comment]
Also note that the source code at ASF do not have `@author` tags in the code, so please remove those.

**Comment by bprager** [comment]
- reformatted, based on @lburgazzoli and @davsclaus requests
- `git merge squash` said: `(nothing to squash)Already up-to-date.`; can you do that when merging?
Thank you guys!

**Comment by davsclaus** [comment]
Had to revert that last commit for this PR to get it merged correctly. Also polished the code as it had some formatting issues. Especially the XML file where we do not use tabs but 2 spaces as indent.

**Comment by bprager** [comment]
Wonderful. Thank you.

**Pull request closed without merging**